src/imagelightbox.js

```diff
@@ -169,8 +173,93 @@
                     $arrows.css('display', 'block');
                 }
             },
+            _pushToHistory = function () {
+                if(!hasHistorySupport || !options.history) {
+                    return;
+                }
+                var value = targets[targetIndex].dataset.ilb2Id;
+                if(!value) {
+                    value = targetIndex;
+                }
+                var query = document.location.search;
+                var newParam = 'imageLightboxIndex=' + value;
+                var newQuery = '?' + newParam;
+
+                if (query) {
+                    var keyRegex = new RegExp('([?&])imageLightboxIndex=[^&]*');
+                    if (query.match(keyRegex) !== null) {
+                        newQuery = query.replace(keyRegex, '$1' + newParam);
+                    } else {
+                        newQuery = query + '&' + newParam;
+                    }
+                }
+                window.history.pushState({imageLightboxIndex: value}, '', newQuery);
+            },
+            _pushQuitToHistory = function () {
+                if(!hasHistorySupport || !options.history) {
+                    return;
+                }
+                var query = document.location.search;
+                if (query) {
+                    var keyRegex1 = new RegExp('[?]imageLightboxIndex=[^&]*');
+                    var keyRegex2 = new RegExp('&imageLightboxIndex=[^&]*');
+                    var newQuery = query.replace(keyRegex1, '?');
+                    newQuery = newQuery.replace(keyRegex2, '');
+                    window.history.pushState({}, '', newQuery);
+                }
+            },
+            _openHistory = function () {
+                if(!hasHistorySupport || !options.history) {
+                    return;
+                }
+                var regex = new RegExp('[?&]imageLightboxIndex(=([^&#]*)|&|#|$)');
+                var results = regex.exec(document.location.search);
+                if (!results) return;
+                if (!results[2]) return;
```

Comment:
Those checks looks weird (and the name "result" is non-describing. What are you checking for here?

Reply:
I am checking whether there is a query parameter (first check) and whether it has a value (second check - it could be `...&imageLightboxIndex=&someOtherOption=...`) It could use a better name though...